interface/TausProducer.h

```diff
@@ -0,0 +1,151 @@
+#ifndef TAUS_PRODUCER
+#define TAUS_PRODUCER
+
+#include <cp3_llbb/Framework/interface/LeptonsProducer.h>
+#include <cp3_llbb/Framework/interface/ScaleFactors.h>
+
+#include <DataFormats/VertexReco/interface/Vertex.h>
+#include <DataFormats/PatCandidates/interface/Tau.h>
+#include <DataFormats/PatCandidates/interface/TauPFSpecific.h>
+
+#include <utility>
+#include <Math/VectorUtil.h>
+
+typedef std::unordered_map<std::string, float> tauDiscriminatorMap;
```

Comment:
Just a matter of style: could you move the typedefs in a namespace (e.g. MatchResult could be private inside TausProducer, union_sets) if they are not stored in the tree, and `Types.h` if they are stored in the output tree?
LorentzVector is defined in [`Types.h`](https://github.com/cp3-llbb/Framework/blob/f59184391fe2ee6b6cc30ecd071ba63e4fa8e242/interface/Types.h#L9) already, you can include it.